core/Pos/Core/Constants.hs

```diff
@@ -80,23 +80,23 @@ isDevelopment = True
 isDevelopment = False
 #endif
 
--- | System start time embeded into binary.
-staticSysStart :: Maybe Timestamp
+-- | System start time embedded into binary.
+staticSysStart :: Timestamp
 staticSysStart
-    | isDevelopment = Nothing
-    | st > 0        = Just $ Timestamp $ sec st
+    | isDevelopment    = error "System start time should be passed \
+                              \as a command line argument in dev mode."
+    | st >= 1400000000 = Timestamp $ sec st
     -- If several local nodes are started within 20 sec,
     -- they'll have same start time
-    | otherwise     = Just $ Timestamp $ sec $
-          (after3Mins `div` divider + alignment) * divider
+    | otherwise        =
+          let frameLength  = st
+              currentFrame = currentPOSIXTime `div` frameLength
+              t = currentFrame * frameLength + (frameLength `div` 2) in
+          Timestamp $ sec t
   where
     st = ccProductionNetworkStartTime coreConstants
-    pause = 30
-    divider = 20
-    after3Mins :: Int
-    after3Mins = pause + unsafePerformIO (round <$> getPOSIXTime)
-    minuteMod = after3Mins `mod` divider
-    alignment = if minuteMod > (divider `div` 2) then 1 else 0
+    currentPOSIXTime :: Int
+    currentPOSIXTime = unsafePerformIO (round <$> getPOSIXTime)
```

Comment:
Dodgy use of `unsafePerformIO`. Can we get rid of it? It means `staticSysStart` depends upon when it's forced.